### ISSUE TITLE
[java] Assert or expectedExceptions

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#2257]
**Affects PMD Version:** 

6.20.0

**Rule:**

New rule: `TestNGAssertOrExpectedExceptions`

**Description:**

TestNG `@Test` annotation has an attribute `expectedExceptions`.  Methods annotated with `@Test` should either have a call to `org.testng.Assert` or `expectedExceptions` should be specified.

Note: There are probably many assertion libraries out there in use.  Please make this new rule configurable to allow other classes than `org.testng.Assert` to be used.

**Code Sample demonstrating the issue:**

PMD should flag a problem with the following method since it does not call `org.testng.Assert` and `@Test` does not have `expectedExceptions`.

```
   @Test
   public static void parseEscapeNoCloseQuote()
   {
      Condition.parseEscaped("\"hello");
   }
```

**Comment by adangel** [comment]
We have a rule like this for JUnit: [JUnitTestShouldIncludeAssert](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert).
We would need the same rule for TestNG. Maybe we can even have one rule for the different test frameworks.

**Comment by Fernal73** [comment]
```java
@Test(dataProvider = "mapProvider",       singleThreaded = true)
```

The rule ignores the above parameters as well while using TestNG and flags the method as not using @Test.

**Comment by lukasgraef** [comment]
I think this issue can be closed by now, as [UnitTestsShouldIncludeAssert](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unittestshouldincludeassert) supports both JUnit and TestNG.

Since PMD 7.19, UnitTestsShouldIncludeAssert also handles TestNG's @Test.expectedException annotation (see https://github.com/pmd/pmd/issues/6188).

**Comment by numeralnathan** [comment]
Agreed.  I am see PMD flagging unit tests without asserts.
